libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_integration.c

```diff
@@ -219,11 +300,63 @@ TEST( Full_BLE_Integration_Test, BLE_Write_Notification_Size_Greater_Than_MTU_3
     }
 }
 
+TEST( Full_BLE_Integration_Test_Connection, BLE_Send_Data_After_Disconected )
+{
+    BTStatus_t xStatus;
+    BLETESTindicateCallback_t xIndicateEvent;
+    BLETESTInitDeinitCallback_t xInitDeinitCb;
+    BLETESTwriteAttrCallback_t xWriteEvent;
+    BLETESTreadAttrCallback_t xReadEvent;
+
+    /* Check communication */
+    xWriteEvent = prvWriteReceive( bletestATTR_SRVCB_CHAR_A, true, false, 0 );
+    prvWriteResponse( bletestATTR_SRVCB_CHAR_A, xWriteEvent, true );
+    xReadEvent = prvReadReceive( bletestATTR_SRVCB_CHAR_A );
+    prvReadResponse( bletestATTR_SRVCB_CHAR_A, xReadEvent, true );
+
+    checkNotificationIndication( bletestATTR_SRVCB_CCCD_E, true );
+    checkNotificationIndication( bletestATTR_SRVCB_CCCD_F, true );
+    prvCheckNotification( true );
+    prvCheckIndication( true );
+
+    /* Disconnect from RPi and Check communication*/
+    prvWaitConnection( false );
+
+    prvWriteResponse( bletestATTR_SRVCB_CHAR_A, xWriteEvent, false );
+    prvReadResponse( bletestATTR_SRVCB_CHAR_A, xReadEvent, false );
+    prvCheckNotification( false );
+    prvCheckIndication( false );
+
+    /* Advertise and Reconnect */
+    prvSetAdvProperty();
+    prvSetAdvData( eBTuuidType128 );
+    prvStartAdvertisement();
+    prvWaitConnection( true );
+
+    /* Check communication */
+    checkNotificationIndication( bletestATTR_SRVCB_CCCD_E, true );
+    checkNotificationIndication( bletestATTR_SRVCB_CCCD_F, true );
+
+    xWriteEvent = prvWriteReceive( bletestATTR_SRVCB_CHAR_A, true, false, 0 );
+    prvWriteResponse( bletestATTR_SRVCB_CHAR_A, xWriteEvent, true );
+    xReadEvent = prvReadReceive( bletestATTR_SRVCB_CHAR_A );
+    prvReadResponse( bletestATTR_SRVCB_CHAR_A, xReadEvent, true );
+    prvCheckNotification( true );
+    prvCheckIndication( true );
+}
+
+/* trigger Adv Stop callback AdvStartCB(with start=false) when Adv timeout. */
+TEST( Full_BLE_Integration_Test, BLE_Integration_Connection_Timeout )
```

Comment:
I am trying to understand this test. Is it like the following:
1. Device starts advertisements
2. Waits for a connection events for BLE_TESTS_SHORT_WAIT seconds
3. Raspberry PI waits at least BLE_TESTS_SHORT_WAIT seconds before connecting to device, so that device times out and test succeeds.

If not I think test can become erratic if Raspberry PI is too fast to connect.

Reply:
> I am trying to understand this test. Is it like the following:
> 
> 1. Device starts advertisement.
> 2. Waits for a connection event for BLE_TESTS_SHORT_WAIT seconds
> 3. Raspberry PI waits at least BLE_TESTS_SHORT_WAIT seconds before connecting to device, so that device times out and test succeeds.
> 
> If not I think test can become erratic if Raspberry PI is too fast to connect.

yes it works in this way.